frontend/src/Tierpfleger.jsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import {useState} from "react";
+import TextField from "@mui/material/TextField";
+import Button from "@mui/material/Button";
+import PflegerAnimalList from "./PflegerAnimalList.jsx";
+import PlegerSelectDialog from "../PflegerSelectDialog.jsx";
+import SuccessSnackbar from "./SuccessSnackbar.jsx";
+
+export default function Tierpfleger() {
+
+    const [PflegerId, setPflegerId] = useState("");
+    const [showTable, setShowTable] = useState(false);
+    const [showDialog, setShowDialog] = useState(false);
+    const [snackBar, setsnackBar] = useState(false);
+    const [animalData, setAnimalData] = useState();
+    const [selectedAnimalData ,setSelectedAnimalData ] = useState();
+    let timeout;
+    
+    async function okDialog(id,columnName,newData){
+        
+        console.log(columnName)
+        console.log(newData)
+        
+        const data = await fetch(`http://localhost:5207/api/Tierpfleger/updateAnimal/${id}&${columnName}&${newData}`)
+        
+        if(data.status == 202){
+            showSnackbar();
+        }
+        
+        setShowDialog(false)
+    }
+    
+    
+    
+
+    async function showAnimals(){
+
+        
+        const data = fetch(`http://localhost:5207/api/Tierpfleger/getAnimal/${PflegerId}`)
+        return (await data).json();
+
+
+    }
+
+
+
+    async function showSnackbar(){
+        function closeSnackbar(){
+            setsnackBar(false);
+        }
+
+        setsnackBar(true);
+        if(timeout){
+            clearTimeout(timeout);
+            timeout=null;
+        }
+
+        timeout = setTimeout(closeSnackbar, 3000);
+
+        return ()=>{
```

Comment:
die returnte Funktion nutzt in diesem Fall nichts, das würde nur innerhalb eines `useEffect` das gewünschte Ergebnis erzielen (das löschen des Timeouts, wenn die Komponente nicht mehr verwendet wird). Du müsstest diesen Teil also in ein `useEffect` auslagern